src/plugins/directoryvalue/directoryvalue_delegate.cpp

```diff
@@ -325,17 +333,19 @@ kdb::KeySet decreaseArrayIndices (kdb::KeySet const & parents, kdb::KeySet const
  */
 KeySetPair splitDirectoriesLeaves (kdb::KeySet const & keys)
 {
+
 	kdb::KeySet leaves;
 	kdb::KeySet directories;
-
-	keys.rewind ();
 	kdb::Key previous;
-	for (previous = keys.next (); keys.next (); previous = keys.current ())
+
+	ssize_t pos = 0;
+	for (previous = keys.at (pos); ++pos < keys.size (); previous = keys.at (pos))
 	{
-		(keys.current ().isBelow (previous) ? directories : leaves).append (previous);
+		(keys.at (pos).isBelow (previous) ? directories : leaves).append (previous);
 	}
```

Comment:
I really don't like this loop it is very hard to understand.

```c
elektraCursor prev = 0;
for (elektraCursor it = 1; it < keys.size(); ++it)
{
    Key * current = keys.at(it);
    Key * previous = keys.at(prev);

    if (current.isBelow(previous))
    {
        directories.append(previous);
    }
    else
    {
        leaves.append(previous);
    }
}
```

I think this is much more readable

Reply:
Thank you for the suggestions!
I adapted the loop to be more readable.